replica/table.cc

```diff
@@ -2883,7 +2868,23 @@ table::seal_snapshot(sstring jsondir, std::vector<snapshot_file_set> file_sets)
     auto out = co_await make_file_output_stream(std::move(f));
     std::exception_ptr ex;
     try {
-        co_await out.write(json.c_str(), json.size());
+        int n = 0;
+        co_await out.write("{\n");
```

Comment:
Instead of preparing the json output by hand, it's better to re-use json formatter from seastar. It has the ability to serialize objects into output_stream<>-s without allocating long strings in memory

Reply:
latest version uses seastar json.

Reply:
It uses json_stream_builder which is fed with elements one-by-one and it appends them to stream with future chains. Why do you tell it doesn't stream?

Reply:
As I commented below, stream_range_as_array can be used to stream directly from the original data to the output file without needing an intermediate container.

Reply:
This only works if the original range us stable over preemption points.

Reply:
> This only works if the original range us stable over preemption points.

Right, and it should be stable at the time we write and seal the manifest.